packages/terra-table/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -0,0 +1,230 @@
+
+const propTypes = {
+  /**
+   * Required string representing a unique identifier for the column header cell.
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * String that specifies the column height. Any valid CSS height value accepted.
+  */
+  headerHeight: PropTypes.string.isRequired,
+
+  /**
+   * String that specifies the default width for columns in the table. Any valid CSS width value is accepted.
+   */
+  width: PropTypes.number.isRequired,
+
+  /**
+   * String of text to render within the column header cell.
+   */
+  displayName: PropTypes.string,
+
+  /**
+   * A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered.
+   * If a `component` value is specified, `sortIndicator` will be ignored. One of `ascending`, `descending`.
+   */
+  sortIndicator: PropTypes.oneOf(Object.values(SortIndicators)),
+
+  /**
+   * Boolean value indicating whether or not the column has an error in the data.
+   */
+  hasError: PropTypes.bool,
+
+  /**
+   * Number that specifies the minimum column width in pixels.
+   */
+  minimumWidth: PropTypes.number,
+
+  /**
+   * Number that specifies the maximum column width in pixels.
+   */
+  maximumWidth: PropTypes.number,
+
+  /**
+   * Boolean value indicating whether or not the column header is selectable.
+  */
+  isSelectable: PropTypes.bool,
+
+  /**
+   * Boolean value indicating whether or not the column header is resizable.
+   */
+  isResizable: PropTypes.bool,
+
+  /**
+   * Height of the parent table.
+   */
+  tableHeight: PropTypes.number,
+
+  /**
+   * The cell's row position in the table. This is zero based.
+   */
+  rowIndex: PropTypes.number,
+
+  /**
+   * The cell's column position in the table. This is zero based.
+   */
+  columnIndex: PropTypes.number,
+
+  /**
+   * Function that is called when a selectable header cell is selected. Parameters:
+   * @param {string} rowId rowId
+   * @param {string} columnId columnId
+   */
+  onColumnSelect: PropTypes.func,
+
+  /**
+   * Function that is called when the mouse down event is triggered on the column resize handle.
+   */
+  onResizeMouseDown: PropTypes.func,
+
+  /**
+   * @private
+   * Object containing intl APIs
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }),
+};
+
+const defaultProps = {
+  hasError: false,
+  isSelectable: false,
+  isResizable: false,
+};
+
+const ColumnHeaderCell = (props) => {
+  const {
+    id,
+    displayName,
+    sortIndicator,
+    hasError,
+    isResizable,
+    isSelectable,
+    tableHeight,
+    width,
+    minimumWidth,
+    maximumWidth,
+    headerHeight,
+    intl,
+    columnIndex,
+    onColumnSelect,
+    onResizeMouseDown,
+    rowIndex,
+  } = props;
+
+  const columnContext = useContext(ColumnContext);
+  const gridContext = useContext(GridContext);
+  const theme = useContext(ThemeContext);
+  const columnHeaderCell = useRef();
+
+  const isGridContext = gridContext.role === GridConstants.GRID;
```

Comment:
Are we able to just made this a function in the GridContext that can be called instead of having to duplicate the logic in multiple files?

Reply:
So it turns out that React contexts are read-only and we can't reference the values of a context inside itself. So This will have to do for now.